src/query/builder.js

```diff
@@ -1061,6 +1061,42 @@ assign(Builder.prototype, {
     return this;
   },
 
+  // Skips locked rows when using a lock constraint.
+  skipLocked() {
+    const { _method } = this;
+    if (!includes(['pluck', 'first', 'select'], _method)) {
+      throw new Error(`Cannot chain .skipLocked() on "${_method}" query!`);
+    }
+    if (!includes(['forShare', 'forUpdate'], this._single.lock)) {
+      throw new Error(
+        '.skipLocked() can only be used after a call to .forShare() or .forUpdate()!'
+      );
+    }
+    if (this._single.waitMode) {
+      throw new Error('.skipLocked() cannot be used with .noWait()!');
```

Comment:
is error still correct if `waitMode === 'skipLocked'`?

Reply:
That's a good point, I'll make the check a bit more explicit.